k9mail/src/main/java/com/fsck/k9/controller/NonQresyncSyncInteractor.java

```diff
@@ -0,0 +1,161 @@
+package com.fsck.k9.controller;
+
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.fsck.k9.Account;
+import com.fsck.k9.mail.Flag;
+import com.fsck.k9.mail.Message;
+import com.fsck.k9.mail.MessageRetrievalListener;
+import com.fsck.k9.mail.MessagingException;
+import com.fsck.k9.mail.store.imap.ImapFolder;
+import com.fsck.k9.mail.store.imap.ImapMessage;
+import com.fsck.k9.mailstore.LocalFolder;
+import timber.log.Timber;
+
+
+class NonQresyncSyncInteractor {
```

Comment:
This covers the CONDSTORE and regular IMAP use cases, since they differ only in the way they handle local flag changes. The QRESYNC has a very different sequence of commands, so it has a separate class.